### ISSUE TITLE
How to compile the engine to a dll so other language can call it?

### ISSUE THREAD

**Issue by mcguu** [lightvector/KataGo#552]
If I use other language such as .net how to compile it to dll so a c# can call it?

**Comment by mega-optimus** [comment]
Instead of as a dll, you can use C# to create a child process running the katago.exe

**Comment by lightvector** [comment]
Yes, just invoke KataGo as a separate process and use the analysis engine whose interface is fully documented here: https://github.com/lightvector/KataGo/blob/master/docs/Analysis_Engine.md

**Comment by mcguu** [comment]
@mega-optimus  what about swift? in mac?

**Comment by HackYardo** [comment]
@mcguu You can use Julia Lang if you have a time of ten days to learn it. By Julia, run and stop katago.exe are as easy as 
```julia
cmd=`./katago.exe mode cfg model`
open(cmd) 
close(cmd)
```

**Comment by y-ich** [comment]
@mcguu san,

I think that [Process](https://developer.apple.com/documentation/foundation/process) in Foundation is for you.
